node/src/service.rs

```diff
@@ -37,11 +37,50 @@ use substrate_prometheus_endpoint::Registry;
 use polkadot_service::CollatorPair;
 use sc_consensus::ImportQueue;
 
+use crate::rpc::{
+	create_full_amplitude, create_full_development, create_full_foucoco, create_full_pendulum,
+	FullDeps, RpcExtension,
+};
+
+pub use amplitude_runtime::RuntimeApi as AmplitudeRuntimeApi;
+pub use development_runtime::RuntimeApi as DevelopmentRuntimeApi;
+pub use foucoco_runtime::RuntimeApi as FoucocoRuntimeApi;
+pub use pendulum_runtime::RuntimeApi as PendulumRuntimeApi;
+
+pub type AmplitudeClient =
+	TFullClient<Block, AmplitudeRuntimeApi, NativeElseWasmExecutor<AmplitudeRuntimeExecutor>>;
+pub type FoucocoClient =
+	TFullClient<Block, FoucocoRuntimeApi, NativeElseWasmExecutor<FoucocoRuntimeExecutor>>;
+pub type PendulumClient =
+	TFullClient<Block, PendulumRuntimeApi, NativeElseWasmExecutor<PendulumRuntimeExecutor>>;
+pub type DevelopmentClient =
+	TFullClient<Block, DevelopmentRuntimeApi, NativeElseWasmExecutor<DevelopmentRuntimeExecutor>>;
+
 type ParachainBlockImport<RuntimeApi, Executor> = TParachainBlockImport<
 	Block,
 	Arc<TFullClient<Block, RuntimeApi, NativeElseWasmExecutor<Executor>>>,
 	TFullBackend<Block>,
 >;
+
+pub trait ParachainRuntimeApiImpl:
```

Comment:
This is a type guard that was used a lot in the functions in this file. Since this type guard is satisfied by all runtime api implementations of each of the four runtimes, we could simplify the type guards by just defining a new trait and then express that the four runtimes implement this trait.